packages/test/test-end-to-end-tests/src/test/cellEndToEndTests.spec.ts

```diff
@@ -231,6 +231,23 @@ describeFullCompat("SharedCell", (getTestObjectProvider) => {
         verifyCellValue(await getCellDataStore(getCellDataStore(Promise.resolve(sharedCell2))), cellValue, 2);
         verifyCellValue(await getCellDataStore(getCellDataStore(Promise.resolve(sharedCell3))), cellValue, 3);
     });
+
+    it("attaches if referring DDS becomes attached or is already attached", async () => {
+        const detachedCell1: ISharedCell = SharedCell.create(dataObject1.runtime);
+        const detachedCell2: ISharedCell = SharedCell.create(dataObject1.runtime);
+
+        // When an unattached cell refers to another unattached cell, both remain unattached
+        detachedCell1.set(detachedCell2.handle);
+        assert.equal(detachedCell1.isAttached(), false, "detachedCell1 is attached");
+        assert.equal(detachedCell2.isAttached(), false, "detachedCell2 is attached");
+
+        // When referring cell becomes attached, the referred cell becomes attached
+        // and the attachment transitively passes to a second referred DDS
+        sharedCell1.set(detachedCell2.handle);
+        assert.equal(sharedCell1.isAttached(), true, "sharedCell1 is not attached");
+        assert.equal(detachedCell1.isAttached(), true, "detachedCell1 is not attached");
+        assert.equal(detachedCell2.isAttached(), true, "detachedCell2 is not attached");
+    });
```

Comment:
I wrote the test for this and it is failing as shown in the console debugger image pasted below. Is this expected behaviour? That the transitively attached cell becomes attached but the middle cell does not become attached?

![1752586e-9642-4eb9-b048-5d252110cd53](https://user-images.githubusercontent.com/6777404/178792373-42e47c82-362f-4d16-86b0-1598f12e42f5.jpg)


cc @jatgarg I hear you are the owner of attachments - can you please let me know if this is expected?

Reply:
Thanks for catching that!